### PR TITLE
Adapted factory reset to CM1 overlay [REVPI-2563]

### PR DESCRIPTION
**Pull request by hjokunbus**
In order to relief the CPU on the older Compute Module, there is now a
distinctive overlay that uses the optimized DWC_OTG USB driver instead of DWC2.
CM1 hereby are autodetected during factory reset.

This PR had to be requested again, since it was formerly made from the wrong branch.

### CODE REVIEW

File: src/revpi-factory-reset.sh [hunk 1]
```diff
@@ -16,15 +18,25 @@ export NEWT_COLORS='root=,black entry=white,black'
...
-		connect-se "RevPi Connect SE" \
-		core "RevPi Core / Core 3(+) / Core S" \
-		flat "RevPi Flat" \
-		3>&1 1>&2 2>&3)
-	if [ "$?" == "1" ]; then
-		return
+	if cm1-detection; then
+		msg="Core CM1 autodetected)
+		ovl=$(whiptail --notags --title "PRODUCT TYPE" --menu "$msg" 0 0 0 \
+			core-cm1 "RevPi Core (autodetected)" \
```
Comment (iluminat23) [inline src/revpi-factory-reset.sh:24]: `core-cm1` is not listed in the `full_devicetype_list`

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: src/revpi-factory-reset.sh (hunk 1 above)
Comment (hjokunbus) [reply src/revpi-factory-reset.sh:24]: Please see line 24

File: src/revpi-factory-reset [hunk 2]
```diff
@@ -8,14 +8,24 @@
...
-full_devicetype_list="compact|connect|core|flat|connect-se"	# all suported devices
+if cm1-detection; then
+	echo "Compute Module 1 has been detected, core-cm1-overlay will be used"
+	echo "automatically."
+	ovl="core-cm1"
+	full_devicetype_list="core-cm1"	# only suported device
+else
+	full_devicetype_list="compact|connect|core|flat|connect-se"	# all suported devices
+fi
+
```
[the remark this answers is not in the document]
Comment (hjokunbus) [reply src/revpi-factory-reset:28]: The requirement was to not give the user any other option than CM1 if one is detected. Perhaps this should be taken through Nico first.

[the remark this answers is not in the document]
Comment (iluminat23) [reply src/revpi-factory-reset:28]: Hmmm, why don't you put the core-cm1 just in the full_..._list, and ignore the autodetect in this case?

File: src/revpi-functions [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+#
+# SPDX-License-Identifier: GPL-2.0
+#
+# Copyright: 2022 Kunbus GmbH
+#
+
+cm1-detection() {
+    grep -m 1 "model name" /proc/cpuinfo | grep -q v6l
+    return $?
+}
```
[the remark this answers is not in the document]
Comment (hjokunbus) [reply src/revpi-functions]: Changed

File: src/revpi-factory-reset (hunk 2 above)
[the remark this answers is not in the document]
Comment (nbuchwitz) [reply src/revpi-factory-reset:28]: It still must be possible to chose from all available device types in the backend script (even on Core with CM1). If a Core was choosen, the detection must be triggered and if positive the respective dtoverlay must be chossen (core-cm1). Also an information message must be presented to the user (like: "NOTE: Core with CM1 was detected. Choosing revpi-core-cm1 instead of revpi-core as device tree overlay ...")

In the frontend (the whiptail stuff) only the Core CM1 should be available (as implemented by you).

[the remark this answers is not in the document]
Comment (hjokunbus) [reply src/revpi-factory-reset:28]: With this change I receive an error that did not occur before on CM1
```
* Failed to apply overlay '0_revpi-core-cm1' (kernel)
```
This leads to an error at the end of revpi-factory-reset.sh, as well. My search indicated that the source is `/opt/vc/bin/dtoverlay` but I still try to figure out why this happened with the suggestion and not without.

[the remark this answers is not in the document]
Comment (zhan-kunbus) [reply src/revpi-factory-reset:28]: this error message should be caused by repeated node labels.

[the remark this answers is not in the document]
Comment (hjokunbus) [reply src/revpi-factory-reset]: That this message somehow created fallout in the calling script was interesting but I have fixed this now.

File: src/revpi-factory-reset.sh [hunk 4]
```diff
@@ -41,7 +52,7 @@ while [ ! -r /home/pi/.revpi-factory-reset ] ; do
 
 	# this creates /home/pi/.revpi-factory-reset on success:
 	/usr/bin/sudo /usr/sbin/revpi-factory-reset "$ovl" "$ser" "$mac" 2>/dev/null
-	if [ "$?" = 1 ]; then
+	if [ "$?" == "1" ]; then
```
Comment (iluminat23) [inline src/revpi-factory-reset.sh:55]: Please put this change in a separate commit!

Comment (iluminat23) [reply src/revpi-factory-reset.sh:55]: You can squash this change into my commit (https://github.com/RevolutionPi/piserial/pull/47/commits/f64e57abbc99babd54ada70d330203c63fe73dbb)  if you like

**Review by iluminat23 — CHANGES_REQUESTED** [review]
Only the small thing about the `if [ "$?" == "1" ]` otherwise it looks good

File: src/revpi-factory-reset.sh (hunk 4 above)
Comment (hjokunbus) [reply src/revpi-factory-reset.sh:55]: No, I won't sqash since this fixes an error I so far only had after my initial commit with CM1 only. This is why I re-ordered the commits after separation.

Comment (nbuchwitz) [reply src/revpi-factory-reset.sh:55]: Is there any special reason for treating this as string and not something like this?

```bash
if [ $? -eq 1 ]; then 
```

Comment (hjokunbus) [reply src/revpi-factory-reset.sh:55]: Not any I can think of. This script obviously has the typical problems of people working too long with systems where /bin/sh points to /bin/bash. As you can see in my commit message of [e91a223](https://github.com/RevolutionPi/piserial/pull/47/commits/e91a22364c589c14f97a34b361a5991e060dccb7), I only changed this for consistency. I otherwise would have rewritten huge parts of the script.

Comment (nbuchwitz) [reply src/revpi-factory-reset.sh:55]: I would be OK with your solution. Just wondering  :smile:

**Review by iluminat23 — APPROVED** [review]
